### PR TITLE
Fixes for Choice editor

### LINKED ISSUE #336: Answers field should always be a JSON array
To keep it consistent with the choices field. If there's a single answer, then it'll just be an array with 1 element.

The generator should output the commas.

### PR DESCRIPTION
**Pull request by Kapin**
Allows multi-select answers to work properly. Needs updates on the front-end still.

Closes #336

### CODE REVIEW

**Comment by Kapin** [comment]
@Trinovantes can you help with the front-end changes or just push a commit to this branch for that?

File: src-django/api/generator.py [hunk 1]
```diff
@@ -107,8 +107,14 @@ def __get_properties(self):
...
-        choice_list = json.loads(self.element.choices)
-        return ','.join(choice_list)
+        return self.__parse(self.element.choices)
+
+    def __parse_answers(self):
+        return self.__parse(self.element.answer)
+
+    def __parse(self, field):
+        item_list = json.loads(field)
+        return ','.join(item_list)
```
Comment (Trinovantes) [inline src-django/api/generator.py]: is there anything in the spec about escaping characters?

Comment (Kapin) [inline src-django/api/generator.py]: No

**Comment by Trinovantes** [comment]
I'll work on this if I have time after conditional logic ui

**Comment by Kapin** [comment]
looks fine to me. Would like someone that isn't us to look at it too.

**Comment by Kapin** [comment]
@connorcimowsky @Ommy plz look

File: src-backbone/app/js/behaviors/sortableBehavior.js [hunk 2]
```diff
@@ -8,14 +8,23 @@ let SortableBehavior = Marionette.Behavior.extend({
...
-
         let $child = ui.item;
         let newIndex = $child.parent().children().index($child);
 
         let collection = this.view.collection;
         let model = collection.get($child.attr('data-model-cid'));
 
+        // Ask user before sorting if it has some side effects
+        if (_.isFunction(this.options.shouldConfirmBeforeSort) && this.options.shouldConfirmBeforeSort(model, newIndex)) {
+            let warningMessage = "Reordering this page will remove all of your conditionals. Are you sure you wish to continue?";
```
Comment (Ommy) [inline src-backbone/app/js/behaviors/sortableBehavior.js]: i18n?

**Comment by Ommy** [comment]
lgtm after comment

**Comment by connorcimowsky** [comment]
same

**Pull request merged**